.github/scripts/create-vm.py

```diff
@@ -0,0 +1,465 @@
+    # fmt: off
+    standard_v2_cpu = [
+        2, 4, 6, 8, 10, 12, 14, 16, 20, 24,
+        28, 32, 36, 40, 44, 48, 52, 56, 60
+    ]
+    # fmt: on
+    if args.platform_id == "standard-v2" and args.cpu not in standard_v2_cpu:
+        raise ValueError(
+            f"CPU cores must be {', '.join(standard_v2_cpu)} for standard-v2 platform"
+        )
+
+    if args.ram % args.cpu != 0:
+        raise ValueError("RAM must be a multiple of CPU cores (1,2,4 etc)")
+
+    if (
+        args.disk_type in ["network-ssd-nonreplicated", "network-ssd-io-m3"]
+        and args.disk_size % 93 != 0  # noqa: W503
+    ):
+        raise ValueError(
+            "Disk size must be a multiple of 93GB for the selected disk type"
+        )
+
+    runner_github_label = generate_github_label()
+
+    user_data, ssh_keys = generate_cloud_init_script(
+        args.user,
+        ssh_keys,
+        args.github_repo_owner,
+        args.github_repo,
+        runner_registration_token,
+        args.github_runner_version,
+        runner_github_label,
+    )
+
+    labels = args.labels
+    labels["runner-label"] = runner_github_label
+
+    request = CreateInstanceRequest(
+        name=args.name,
+        folder_id=args.folder_id,
+        zone_id=args.zone_id,
+        platform_id=args.platform_id,
+        resources_spec=ResourcesSpec(
+            memory=args.ram * 1024 * 1024 * 1024, cores=args.cpu, core_fraction=100
+        ),
+        boot_disk_spec=AttachedDiskSpec(
+            auto_delete=True,
+            mode=AttachedDiskSpec.Mode.READ_WRITE,
+            disk_spec=AttachedDiskSpec.DiskSpec(
+                type_id=args.disk_type,
+                size=args.disk_size * 1024 * 1024 * 1024,
+                image_id=args.image_id,
+            ),
+        ),
+        labels=labels,
+        network_interface_specs=[
+            NetworkInterfaceSpec(
+                subnet_id=args.subnet_id,
+                primary_v4_address_spec=PrimaryAddressSpec(
+                    one_to_one_nat_spec=OneToOneNatSpec(ip_version=IpVersion.IPV4)
+                ),
+            )
+        ],
+        metadata={
+            "user-data": user_data,
+            "ssh-keys": "\n".join([f"{args.user}:{key}" for key in ssh_keys]),
+            "serial-port-enable": "1",
+        },
+    )
+
+    # Create the VM
+    if args.apply:
+        result = sdk.create_operation_and_get_result(
+            request=request,
+            service=InstanceServiceStub,
+            method_name="Create",
+            response_type=Instance,
+            meta_type=CreateInstanceMetadata,
+            timeout=args.timeout,
+            logger=logging.getLogger(),
+        )
+        instance_id = result.response.id
+        name = result.response.name
+        primary_ipv4 = result.response.network_interfaces[0].primary_v4_address
+        external_ipv4 = primary_ipv4.one_to_one_nat.address
+        local_ipv4 = primary_ipv4.address
+        if instance_id:
+            logging.info(
```

Comment:
Use logging parametrization instead of an f string. I know it is not a big deal for now, but string-formatting-before logging is considered a bad practice overall. https://www.palkeo.com/en/blog/python-logging.html